utils/build_rpms_from_scratch.sh

```diff
@@ -6,52 +6,126 @@ set -uex
 
+# Get distro information so can be used throughout the script globally
+. /etc/os-release
+OS_NAME=$NAME
+OS_VERSION=$VERSION_ID
+OS_ID=$ID
+
+# -----------------------------------------------------------------------------
+# This first set of functions serve to abstract away some OS specifics. It is
+# assumed that by the time these functions are called, the OS and Version
+# have been verified
+# -----------------------------------------------------------------------------
+
 # Set up the temporary local repository configuration with GPG check disabled
 setup_local_repo() {
   local rpms_dst=$1
 
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+    repo_file=/etc/zypp/repos.d/local-rpms.repo
+  else
+    repo_file=/etc/yum.repos.d/local-rpms.repo
+  fi
+
   echo "[local-rpms]
 name=Local RPM Repository
 baseurl=file://$rpms_dst
 enabled=1
-gpgcheck=0" > /etc/yum.repos.d/local-rpms.repo
+gpgcheck=0" > "$repo_file"
 }
 
 # Clean up the temporary repository configuration
 cleanup_local_repo() {
-  rm -f /etc/yum.repos.d/local-rpms.repo
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+    rm -f /etc/zypp/repos.d/local-rpms.repo
+  else
+    rm -f /etc/yum.repos.d/local-rpms.repo
+  fi
 }
 
-dnf_builddep() {
-  local rpms_dst=$1
-  local spec_file="$2"
-  # dnf --nogpgcheck --repofrompath rpms,"$rpms_dst" --refresh builddep -y "$spec_file"
-  dnf --refresh builddep -y "$spec_file"
+install_dependencies() {
+  local spec_file="$1"
+
+  if [[ "$OS_ID" == "opensuse-leap" ]]; then
+   dependencies=$(rpmspec -q --buildrequires "$spec_file" | grep -v "^rpmlib" | grep -v "^/" | tr '\n' ' ')
+      if [ -n "$dependencies" ]; then
+          # Install dependencies using zypper
+          echo "Installing build dependencies: $dependencies"
+          zypper install --allow-vendor-change -y $dependencies
+      else
+          echo "No build dependencies found in the spec file."
+      fi
+  else
+    dnf --refresh builddep -y "$spec_file"
+  fi
+}
+
+prepare_env_leap() {
+  suse_version=$OS_VERSION # default to '15.4'
+
+  # Update and install required packages using zypper
+  zypper refresh
+
+  # Install needed packages
+  # git so can clone the repo
+  # rpm-build, rpmdevtools, createrepo for building the rpms and setting up the rpm repo
+  # make is used for managing the rpmbuild process. This should probably be a dependency of the rpm.spec files, but not
+  # python3-Sphinx is needed by dpdk. The file /usr/bin/sphinx-build is listed as a dependency in the spec file,
+  #   however, zypper isn't able to resolve a filename to a package and the scripting to handle that case is pretty
+  #   messy, especially because "zypper what-provides" actually returns three options for sphinx-build. So, forgoing
+  #   automation here and simply hardcoding the dependency.
+  zypper install -y \
+      git \
+      rpm-build \
+      rpmdevtools \
+      createrepo_c \
+      make \
+      python3-Sphinx
```

Comment:
Any idea what are requiring this?  If it's failing `rpmbuild`s that's a packaging bug.

Reply:
There's a comment for python3-Sphinx:
  # python3-Sphinx is needed by dpdk. The file /usr/bin/sphinx-build is listed as a dependency in the spec file,
  #   however, zypper isn't able to resolve a filename to a package and the scripting to handle that case is pretty
  #   messy, especially because "zypper what-provides" actually returns three options for sphinx-build. So, forgoing
  #   automation here and simply hardcoding the dependency.
I recognize it's not ideal, but weighing the effort and complexity (at least for me) for managing the dependency dynamically compared to hardcoding it and dealing with the impact on maintainability, I chose this approach. Definitely  open to suggestions.

Reply:
So that `dpdk` needs it means that the `BuildRequires:` in `dpdk.spec` should specify it.  Did you get a failure to build `dpdk` without it?  If you had a link to the failure that'd be awesome.

Reply:
`BuildRequires:` specifies `/usr/bin/sphinx-build`, however, zypper doesn't resolve filenames to packages to install.

Reply:
The other approach is that even if you want to continue to use `zypper` for the bulk of this script, recognize that `zypper` is inferior (or we are just ignorant) in regards to installing build-dependencies and install and use `dnf` for that.  Perhaps if we are more obvious about this short-coming (or our ignorance) in `zypper`, somebody with more knowledge of `zypper` that is using and looking at this script can find/suggest a better way.

But I think by simply adding `python3-Sphinx` as an exception to the package install list here we are making the issue much (much!) more subtle than if we were to use `dnf` with a nice big comment about why we are using it.